core/domain/user_services.py

```diff
@@ -607,9 +623,11 @@ def get_user_settings_by_gae_id(gae_id, strict=False):
     Raises:
         Exception. The value of strict is True and given gae_id does not exist.
     """
-    user_settings_model = user_models.UserSettingsModel.get_by_gae_id(gae_id)
-    if user_settings_model is not None:
-        user_settings = _transform_user_settings(user_settings_model)
+    user_auth_details_model = user_models.UserAuthDetailsModel.get_by_auth_id(
+        feconf.AUTH_METHOD_GAE, gae_id)
+    if user_auth_details_model is not None:
+        user_settings = _transform_user_settings(
```

Comment:
Just to check: when a user settings model is deleted, do we ensure that the user auth details model is also deleted?

Reply:
Yes I have updated the wipeout code to do that.